### DISCUSSION TITLE
wait for a transaction db.tx

### DISCUSSION THREAD

**Discussion by davidanrod** [vitaly-t/pg-promise#849] [Q&A]
Hi,

after reading the docs and seeing the examples, i didn't realized how to wait for a transaction to continue the execution of the program.. 

the basic idea it's something like this,

```
db.tx( async t => { 
//do something
})

//execute only after transaction
```

This is probably a noob question since nodejs it's not my world, and i probably should have more knowladge about promises and async/wait, but if some of you could help at this step, would be nice..

Best,
David.

**Comment by vitaly-t** [comment]
It is a noob question, yes. You simply use `await`, to wait for the transaction to finish:

```js
await db.tx( async t => { 
    //do something
})

// execute only after transaction
```
